docs/architecture/notifications.md

```
@@ -0,0 +1,922 @@
+    EXTRACT --> LOOP{For each subscriber}
+    LOOP --> CHECK{sender == subscriber?}
+    CHECK -->|Yes| SKIP[Skip delivery]
+    CHECK -->|No| NEXT[Continue to mention filter]
+```
+
+Each adapter extracts the sender with a single field lookup:
+- Slack: `event.User` or `event.BotID`
+- Telegram: `message.From.UserName`
+- Discord: `message.Author.ID`
+
+The sender string is stripped of the `[platform] ` prefix (e.g., `[slack] eng-01` becomes `eng-01`) before comparison.
```

markdownlint-cli2 (0.22.0)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)